app/scripts/models/mixins/resume-token.js

```diff
@@ -44,16 +54,47 @@ define(function (require, exports, module) {
     },
 
     /**
-     * Sets model properties from a ResumeToken.
+     * Sets model properties from a ResumeToken, unless said token
+     * is invalid according to `this.resumeTokenSchema`.
      *
      * @method populateFromResumeToken
      * @param {ResumeToken} resumeToken
      */
     populateFromResumeToken: function (resumeToken) {
       if (this.resumeTokenFields) {
-        this.set(resumeToken.pick(this.resumeTokenFields));
+        var pickedResumeToken = resumeToken.pick(this.resumeTokenFields);
+
+        if (isResumeTokenValid.call(this, pickedResumeToken)) {
+          this.set(pickedResumeToken);
+        }
       }
     }
   };
+
+  function isResumeTokenValid (resumeToken) {
+    if (this.resumeTokenSchema) {
+      var error = vat.validate(resumeToken, this.resumeTokenSchema).error;
+
+      if (error) {
+        if (this.sentryMetrics) {
+          if (error instanceof ReferenceError) {
+            error = authErrors.toMissingResumeTokenPropertyError(error.key);
+          } else {
+            error = authErrors.toInvalidResumeTokenPropertyError(error.key);
+          }
+
+          // HACK: Interpolate the invalid property name into the error
+          //       message. One day this will be handled automagically.
+          error.message = authErrors.toInterpolatedMessage(error);
```

Comment:
Does https://github.com/mozilla/fxa-content-server/pull/3682/commits/91923ca7f0fc7aec0461cc15d17e33aa2423c368 make it better?